### ISSUE TITLE
Cleaned-up version of QFusionStyle

### ISSUE THREAD

**Issue by JulienMaille** [randrew/phantomstyle#6]
Good job with this phantomstyle Andrew, there's not many QStyle floating around. I'm unfortunately still stuck with Qt4.8 so I can't really play with phantomstyle.

I was wondering if there's anywhere where you commited your initial fixes to QFusionStyle, for us to learn what you cleaned up (I'm referring to the wrong sizing, useless repaint and so on you describe in the readme)

**Comment by randrew** [comment]
Unfortunately, there is not :(

I went through the whole file and replaced the code in big chunks. The git history for this was in a separate temp repo with very dirty history (and proprietary test/example code) that I threw away -- it wouldn't have been useful. When I found something that was wrong, I would make a note of it and then write my own version without that problem.

Lots of the stuff that was wrong with QFusionStyle was tied to how it would draw things. For example, it draws a semi-transparent gradient in the scrollbar gutter in order to add a depth/shadowing effect. But it does several things incorrectly (two serious math errors partially cancelling out into only a minor error), which causes some artifacts to appear in the corner of the gutter. You can notice it more easily with a dark palette, but it's still there with a light palette. PhantomStyle doesn't use a semi-transparent gradient on top of a base color -- it draws it by filling the pixels only once, using rectangle fills. So while it was possible for me to record why QFusionStyle was doing something incorrectly, the "fix" in PhantomStyle was just a completely different implementation.

Basically, the fixes for QFusionStyle never existed in a state where they had been "fixed" but not "changed" into PhantomStyle. I went directly from one to the other, without a "fixed but not changed" step.

**Comment by randrew** [comment]
I should also point out that the code for QFusionStyle is significantly different in Qt4 compared to Qt5 -- even if there existed a "fixed" version of QFusionStyle before it was turned into PhantomStyle, it probably would be too different from the Qt4 version to be useful. (Lots of code for widgets that don't exist in Qt4, QPainter stuff that has changed, DPI scaling code, etc.)

**Comment by randrew** [comment]
Closing due to no response and no further actions to be taken on this open-ended request.

**Comment by JulienMaille** [comment]
Thanks a lot for your detailed answer Randrew, I'm right now in holidays and that's why I was a bit slow to reply.
